modules/4337/contracts/Safe4337Module.sol

```diff
@@ -99,14 +135,16 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
         // The entry point address is appended to the calldata in `HandlerContext` contract
         // Because of this, the relayer may manipulate the entry point address, therefore we have to verify that
         // the sender is the Safe specified in the userOperation
-        require(safeAddress == msg.sender, "Invalid caller");
+        if (safeAddress != msg.sender) {
+            revert InvalidCaller();
+        }
 
         // We check the execution function signature to make sure the entry point can't call any other function
         // and make sure the execution of the user operation is handled by the module
-        require(
-            this.executeUserOp.selector == bytes4(userOp.callData) || this.executeUserOpWithErrorString.selector == bytes4(userOp.callData),
-            "Unsupported execution function id"
-        );
+        bytes4 selector = bytes4(userOp.callData);
+        if (selector != this.executeUserOp.selector && selector != this.executeUserOpWithErrorString.selector) {
```

Comment:
Question: Would using `userOp.callData` directly be cheaper here than using a `bytes4 memory`?

Reply:
`bytes4` is a stack variable, so there is no memory involved. Roughly speaking, `bytes4(userOp.callData)` needs to do:
- Check that the length of the `bytes calldata` is enough (so a branch)
- Push the value of the calldata offset onto the stack
- `calldataload` at the offset of the `bytes calldata`
- Mask the loaded word with `0xffffffff00000000000000000000000000000000000000000000000000000000`

While storing it as a stack variable makes the second access just be a `dup` opcode.